rules/CodingStyle/Application/UseImportsAdder.php

```diff
@@ -74,42 +77,48 @@ public function addImportsToStmts(
                 break;
             }
 
-            $nodesToAdd = array_merge([new Nop()], $newUses);
+            $position = $key + 1;
 
-            $this->mirrorUseComments($stmts, $newUses, $key + 1);
+            $useComments = $this->getAndRemoveExistingUseComments($stmts, $position);
 
             // remove space before next use tweak
-            if (isset($stmts[$key + 1]) && ($stmts[$key + 1] instanceof Use_ || $stmts[$key + 1] instanceof GroupUse)) {
-                $stmts[$key + 1]->setAttribute(AttributeKey::ORIGINAL_NODE, null);
+            if (isset($stmts[$position]) && ($stmts[$position] instanceof Use_ || $stmts[$position] instanceof GroupUse)) {
+                $stmts[$position]->setAttribute(AttributeKey::ORIGINAL_NODE, null);
             }
 
-            array_splice($stmts, $key + 1, 0, $nodesToAdd);
+            array_splice($stmts, $position, 0, [new Nop()]);
+            ++$position;
 
-            $fileWithoutNamespace->stmts = $stmts;
-            $fileWithoutNamespace->stmts = array_values($fileWithoutNamespace->stmts);
+            $fileWithoutNamespace->stmts = $this->insertUseNodesInStatements($stmts, $newUses, $position, false);
```

Comment:
Now instead of just adding the new `Use_` nodes we call this new function which will take care of inserting them, either unsorted as before or in sorted order